Logic/igtlioConnector.h

```diff
@@ -339,8 +354,8 @@ class OPENIGTLINKIO_LOGIC_EXPORT igtlioConnector : public vtkIGTLIOObject
   vtkMutexLockPointer                     EventQueueMutex;
 
   // Collect commands before they enter the circular buffer, in order to make sure that they are not overwritten
-  std::queue<igtl::MessageBase::Pointer>  CommandQueue;
-  vtkMutexLockPointer                     CommandQueueMutex;
+  std::queue<std::pair<int, igtl::MessageBase::Pointer> >  IncomingCommandQueue;
```

Comment:
For code clarity, could you typedef int to whatever this means? I think it's client id? `typdedef int ClientIDType`

Reply:
In general, usage of std::pair should be avoided. It is almost as simple to add a struct that stores two values, but the struct has the advantages of accessing members by meaningful names (instead of just "first" and "second") and easily extensible to have more members as needed.